scripts/build-controller-image.sh

```diff
@@ -70,15 +71,24 @@ if [[ $QUIET = "false" ]]; then
     echo " git commit: $SERVICE_CONTROLLER_GIT_COMMIT"
 fi
 
+# if workspace has local 'runtime' module
+# then use Dockerfile which allows references to local module from service controller go.mod file
+if [[ -d "${ACK_DIR}/runtime" ]]; then
+  DOCKER_BUILD_CONTEXT="$ACK_DIR"
+  DOCKERFILE="${ROOT_DIR}"/Dockerfile.localmodules
+else
+  DOCKER_BUILD_CONTEXT="$SERVICE_CONTROLLER_SOURCE_PATH"
+fi
```

Comment:
The reasons I'm saying this are:
- Building the official controller images should not use the local `runtime` repository, but rather a specific published version/commit.
- To develop controllers we will be, most likely, doing changes to the `controller-gen` and rarely having to change something in `runtime`.
- It is possible to forget a local repository unsynced/out of date/unrebased, which might be a little hard to debug if users are not aware of this behaviour.

Reply:
Agree with @A-Hilaly here. The local development option should be explicitly-stated.

@kumargauravsharma, if you want to create some additional helper Makefile targets like `local-kind-test` that set `LOCAL=true` environment variable, that's cool.

Reply:
Considering that
> It is upto the `elasticache-controller/go.mod` to use runtime module from local workspace or from its github version; `Dockerfile.localmodules` just makes the local `runtime` directory available during built.

The use of local vs remote `runtime` is controlled by the consumer module `elasticache-controller/go.mod` file and not by `Dockerfile.localmodules`.

Please review.

Reply:
If i understand your point @kumargauravsharma the `go.mod` might contain a `replace` statement or not and that will control which version/commit of `runtime` is used.

However i still don't get why would we change the Dockerfile and build context, just because a repository is cloned locally.. This way we can never use the normal Dockerfile just because we have local `runtime` clone.. and if we ever want to release the official controller images we will have to delete the local `runtime` clone.

Reply:
Another point to take in consideration is that, the build time using `Dockerfile.local` is defintly higher than the build time using `Dockerfile` because:
- We run extra `go mod download` commands
- the Docker daemon have to load more files that necessary because the Build context used for `Dockerfile.local` is way bigger than the one used for `Dockerfile`

Reply:
IMO, ideally the default behavior should be using the normal `Dockerfile`, and only switch to `Dockerfile.local` if `../runtime` exists and `LOCAL=true`